src/gentropy/dataset/test.py

```diff
@@ -0,0 +1,55 @@
+"""Study locus dataset."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pyspark.sql.types import (
+    DoubleType,
+    LongType,
+    StringType,
+    StructField,
+    StructType,
+)
+
+from gentropy.dataset.dataset import Dataset
+
+
+@dataclass
+class TestClass(Dataset):
+    """Plain test class demonstrating the use of automatic  ID generation and instantiation without schema."""
+
+    # This list defines the uniqueness of the dataset - these have to be mandatory fields:
+    _unique_fields = ["a", "b", "c"]
+    _id_column = "testId"
+
+    def __post_init__(self: Dataset) -> None:
+        """Class specific post-init, overwriting the parent method."""
+        # Upon initialising, we assign schema - this is based on the class method, so it would work without instantiating the class:
+        self._schema = self.get_schema()
+
+        # Only calculating the identifier if it is not present in the dataframe already:
+        if "testId" not in self._df.columns:
+            self._df = self._df.withColumn(
+                self._id_column, self._generate_identifier(self._unique_fields)
```

Comment:
We can call the `_generate_identifier` in the Dataset.__post_init__ with the fields that are inherited from the child class. Then the only things to consider are:
1. if dataset requires the index field
2. the index field name
3. the fields that build index field
Defining the above in the child and will result in useage of them, but returning to the parent class __post_init__ to run the method itself to generate the index.

Reply:
But we don't need identifier in every dataset. We do, however want a list of columns defining uniqueness though.